src/environmentd/tests/server.rs

```diff
@@ -1436,6 +1450,16 @@ fn test_old_storage_usage_records_are_reaped_on_restart() {
         let server = harness.clone().start_blocking();
         let mut client = server.connect(postgres::NoTls).unwrap();
         // Create a table with no data, which should have some overhead and therefore some storage usage
+        let mut mz_client = server
+            .pg_config_internal()
+            .user(&SYSTEM_USER.name)
+            .connect(postgres::NoTls)
+            .unwrap();
+        // Otherwise fails with "Records were not reaped!"
+        mz_client
+            .batch_execute("ALTER SYSTEM SET storage_source_decode_fuel = 100000")
```

Comment:
This setting really shouldn't change the output of anything AIUI... maybe there's a timing issue?

My understanding is that these records are not reaped synchronously, so it seems like there could be some race here, but I'm not really an expert on this code. Maybe @jkosh44?

Reply:
Yeah this parameter should have no effect on records reaped. Increasing this will increase the likelihood of compacting already reaped records. We're supposed to read records only during startup. I agree with @bkirwi there's likely some other race going on.

Reply:
This test is notoriously flakey and relies on background work. I would increase the timeout on line 1471 from 5s to 30s.

Reply:
That sounds good to me!

Reply:
Increasing the timeout did not work! I'll open a separate issue for us to dig deeper into this when time allows.